recipes/sqlcipher/all/conanfile.py

```diff
@@ -108,122 +118,94 @@ def _build_visual(self):
-            autotools_env.make(args=["install"])
+        shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
+                    os.path.join(self._source_subfolder, "config.sub"))
+        shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS,
+                    os.path.join(self._source_subfolder, "config.guess"))
+        configure = os.path.join(self._source_subfolder, "configure")
+        self._chmod_plus_x(configure)
+        if self.settings.os == "Macos":
+            tools.replace_in_file(configure, r"-install_name \$rpath/", "-install_name ")
+        autotools = self._configure_autotools()
+        autotools.make()
 
-    @property
-    def _arch_id_str_compiler(self):
-        return {"x86": "i686",
-                "armv6": "arm",
-                "armv7": "arm",
-                "armv7hf": "arm",
-                # Hack: config.guess of sqlcipher does not like aarch64
-                "armv8": "armv8",
-                "mips64": "mips64"}.get(str(self.settings.arch),
-                                        str(self.settings.arch))
-
-    def _get_configure_args(self, absolute_install_dir):
-        args = [
-            "--prefix=%s" % absolute_install_dir,
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
 
-            self._autotools_bool_arg("shared", self.options.shared),
-            self._autotools_bool_arg("static", not self.options.shared),
-            "--enable-tempstore=%s" % self._temp_store_autotools_value,
+        yes_no = lambda v: "yes" if v else "no"
+        args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--enable-tempstore={}".format(self._temp_store_autotools_value),
             "--disable-tcl",
         ]
         if self.settings.os == "Windows":
             args.extend(["config_BUILD_EXEEXT='.exe'", "config_TARGET_EXEEXT='.exe'"])
-        if self._use_commoncrypto():
-            args.extend(["--with-crypto-lib=commoncrypto"])
-        return args
 
-    def _autotools_bool_arg(self, arg_base_name, value):
-        prefix = "--enable-" if value else "--disable-"
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        if self.settings.os == "Linux":
+            self._autotools.libs.append("dl")
+            if not self.options.with_largefile:
+                self._autotools.defines.append("SQLITE_DISABLE_LFS=1")
+        self._autotools.defines.append("SQLITE_HAS_CODEC")
 
-        return prefix + arg_base_name
+        env_vars = self._autotools.vars
+        tclsh_cmd = self.deps_env_info.TCLSH
+        env_vars["TCLSH_CMD"] = tclsh_cmd.replace("\\", "/")
+        if self._use_commoncrypto():
+            env_vars["LDFLAGS"] += " -framework Security -framework CoreFoundation "
```

Comment:
don't like this kind of injection, but seems like there is no better way...

Reply:
Yes, don't like it either, but it was already there, and there are already a lot of modifications. Might be improved later.